### PR TITLE
Upgrade Intel-classic compiler in macOS CI

### PR DESCRIPTION
**Pull request by perazz**
Address #776: 2021.4 -> 2023.1

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -169,7 +169,7 @@ jobs:
         cxx: [icpc]
     env:
       MACOS_HPCKIT_URL: >-
-        https://registrationcenter-download.intel.com/akdlm/irc_nas/18242/m_HPCKit_p_2021.4.0.3389_offline.dmg
+        https:/registrationcenter-download.intel.com/akdlm/IRC_NAS/a99cb1c5-5af6-4824-9811-ae172d24e594/m_HPCKit_p_2023.1.0.44543.dmg
```
Comment (jvdp1) [inline .github/workflows/CI.yml:172]: Should it be:
```suggestion
        https://registrationcenter-download.intel.com/akdlm/IRC_NAS/a99cb1c5-5af6-4824-9811-ae172d24e594/m_HPCKit_p_2023.1.0.44543.dmg
```

**Review by jvdp1 — APPROVED** [review]
LGTM! Thank you for the update. Pending the minor comment, it can be merged.

For my knowledge: where did you find this path/link?

File: .github/workflows/CI.yml (hunk 1 above)
Comment (perazz) [reply .github/workflows/CI.yml:172]: Sorry I hadn't noticed that! I did a copy-and-paste from the @fortran-lang setup-fortran repository: 

https://github.com/fortran-lang/setup-fortran/blob/2a11c57ecf26d40ea43d66a33f9f6fc4d593d6f8/setup-fortran.sh#L402

Comment (perazz) [reply .github/workflows/CI.yml:172]: I will open a PR on that too, although it works, it looks like a typo

**Comment by jvdp1** [comment]
@perazz : I forgot to ask to upate the [README.md](https://github.com/fortran-lang/stdlib/blob/8853ecdea0de04db9bcb73ab285ff6ec8153e909/README.md?plain=1#L95), please?

**Comment by jvdp1** [comment]
Thank you @perazz . I'll merge this PR as it is a small PR related only to the CI.

**Pull request merged**